pages/gallery/gallery-animation.vue

```diff
@@ -0,0 +1,169 @@
+<template>
+	<canvas ref="canvasRef"></canvas>
+</template>
+
+<script>
+export default {
+	name: "gallery-animation",
+	props: {
+		src: {
+			type: String,
+			required: true,
+		},
+		mcmeta: {
+			type: Object,
+			default: () => ({ animation: {} }),
+		},
+		// Determine if the image is a tiled texture (used for flowing fluids which are 2x2 textures)
+		isTiled: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	data() {
+		return {
+			canvasRef: null,
+			image: null,
+			sprites: [],
+			frames: {},
+			currentTick: 1,
+			tickingRef: null,
+			updateCanvasTimeout: null,
+		};
+	},
+	methods: {
+		loadImage() {
+			const img = new Image();
+			img.setAttribute("crossorigin", "anonymous");
+			img.src = this.src;
+
+			img.onload = () => {
+				this.image = img;
+				this.tickingRef = setInterval(() => {}, 1000 / 20);
+			};
+
+			img.onerror = () => {
+				this.image = null;
+				if (this.tickingRef) {
+					clearInterval(this.tickingRef);
+					this.tickingRef = null;
+				}
+			};
+		},
+		calculateFrames() {
+			if (!this.image || !this.mcmeta) return;
+
+			const animation = this.mcmeta.animation ?? {};
+			const animationFrames = [];
+
+			if (animation.frames) {
+				for (const frame of animation.frames) {
+					if (typeof frame === "object") {
+						animationFrames.push({ index: frame.index, time: Math.max(frame.time, 1) });
+					} else {
+						animationFrames.push({ index: frame, time: animation.frametime ?? 1 });
+					}
+				}
+			} else {
+				const framesCount = this.isTiled
+					? this.image.height / 2 / (this.image.width / 2)
+					: this.image.height / this.image.width;
+				for (let fi = 0; fi < framesCount; fi++) {
+					animationFrames.push({ index: fi, time: animation.frametime ?? 1 });
+				}
+			}
+
+			const framesToPlay = {};
+			let ticks = 1;
+			animationFrames.forEach((frame, index) => {
+				for (let t = 1; t <= frame.time; t++) {
+					framesToPlay[ticks] = [[frame, 1]];
```

Comment:
use object instead of tuple for more obvious destructuring
```suggestion
					framesToPlay[ticks] = [{ data: frame, alpha: 1 }];
```